sdk/src/main/java/com/hedera/hashgraph/sdk/EthereumFlow.java

```diff
@@ -0,0 +1,133 @@
+package com.hedera.hashgraph.sdk;
+
+import java.util.Arrays;
+
+import java8.util.concurrent.CompletableFuture;
+
+import java.util.concurrent.TimeoutException;
+import javax.annotation.Nullable;
+
+public class EthereumFlow implements WithExecute<TransactionResponse> {
+    @Nullable
+    private EthereumTransactionData ethereumData;
+
+    @Nullable
+    private FileId callDataFileId;
+
+    @Nullable
+    private Hbar maxGasAllowance;
+
+
+    public EthereumFlow() {
+    }
+
+    @Nullable
+    public EthereumTransactionData getEthereumData() {
+        return ethereumData;
+    }
+
+    public EthereumFlow setEthereumData(byte[] ethereumData) {
+        this.ethereumData = EthereumTransactionData.fromBytes(ethereumData);
+        return this;
+    }
+
+    @Nullable
+    public Hbar getMaxGasAllowance() {
+        return maxGasAllowance;
+    }
+
+    public EthereumFlow setMaxGasAllowance(Hbar maxGasAllowance) {
+        this.maxGasAllowance = maxGasAllowance;
+        return this;
+    }
+
+    private static FileId createFile(byte[] callData, Client client) throws PrecheckStatusException, TimeoutException {
+        try {
+            var transaction = new FileCreateTransaction()
+                    .setContents(Arrays.copyOfRange(callData, 0, Math.min(4096, callData.length)))
```

Comment:
Set static fields for `4096` and `5120`
Names should convey the relevance of the byte size threshold

Reply:
That's fair, also I should double check the 5120 is correct; for some reason I remember that number, but don't know where it came from.